src/constructs/aws/ServerSideWebsite.ts

```diff
@@ -115,11 +116,23 @@ export class ServerSideWebsite extends AwsConstruct {
         })();
         const backendCachePolicy = CachePolicy.CACHING_DISABLED;
 
-        const apiId =
-            configuration.apiGateway === "rest"
-                ? this.provider.naming.getRestApiLogicalId()
-                : this.provider.naming.getHttpApiLogicalId();
-        const apiGatewayDomain = Fn.join(".", [Fn.ref(apiId), `execute-api.${this.provider.region}.amazonaws.com`]);
+        // Resolve domain to use as cloudfront origin
+        const originDomain = (() => {
+            if (configuration.functionName !== void 0) {
```

Comment:
I wonder if this could be optional: if the config doesn't use API Gateway, and exactly 1 function has `url: true`, we could auto-select it?

Reply:
What we need to do if there are multiple functions using url or api gateway.

For example:
1. function "web" uses url
2.function "api" uses apigw

Thats the reason behind the new "functionName", to be able to select which function we will get the url.

With APIGW this doesn't happen because when you have multiple functions with apigw only one "HttpApi" is created.

Reply:
I definitely agree with the new option, it's just that if there's only 1 function with `url: true` then it could be automatic?

(simple defaults)

Reply:
I agree with that.

If there is exactly one function defined, and it has `url: true` we must use that function lambda url by default

Reply:
Maybe we can add a new config parameter to define wich endpoint to use if both are defined?
- `api-gateway`
- `lambda-url`

Reply:
Oh I would add another condition: no API Gateway is defined.

But I'm not sure how easy it is to check 🤔 FYI maybe this code will help just a bit: https://github.com/serverless/serverless/blob/f22354e8dd45871915e21e5656055a9213f93ff0/lib/plugins/aws/info/display.js#L35 It's the code that displays the URL of the deployed application, and if there's only 1 it shows `endpoint: ...` instead of `endpoints: ...`

Reply:
Added some checks